cloudSupport/src/main/scala/cromwell/cloudsupport/auth/AuthMode.scala

```diff
@@ -0,0 +1,38 @@
+package cromwell.cloudsupport.auth
+
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.auth.Credentials
+import cromwell.cloudsupport.auth.AuthMode.OptionLookup
+import org.slf4j.LoggerFactory
+
+object AuthMode {
+
+  class OptionLookupException(val key: String, cause: Throwable) extends RuntimeException(key, cause)
+
+  type OptionLookup = String => String
+  type CredentialsValidation = Credentials => Unit
+
+  lazy val jsonFactory = JacksonFactory.getDefaultInstance
+
+}
+
+trait AuthMode[A] {
+
+  lazy val jsonFactory = JacksonFactory.getDefaultInstance
+  lazy val log = LoggerFactory.getLogger(getClass.getSimpleName)
```

Comment:
Could you, please, explain, why should logger be lazy?

Reply:
From my answer to @ilyaonishenko, it is more appropriately to define logger at AuthMode's object. So then lazy val seems like a good because it used only once at one of the authentification google modes.